tests/unit/checkboxradio/checkboxradio_options.js

```diff
@@ -0,0 +1,180 @@
+		strictEqual( checkbox.is( ":disabled" ), true,
+			"checkbox is disabled when option is set to true" );
+
+		checkbox.checkboxradio( "destroy" );
+		checkbox.prop( "disabled", true );
+		checkbox.checkboxradio();
+
+		strictEqual( widget.hasClass( "ui-state-disabled" ), true,
+			"label gets ui-state-disabled when checkbox is disabled in dom on startup" );
+		strictEqual( checkbox.is( ":disabled" ), true,
+			"checkbox is disabled when checkbox is disabled in dom on startup" );
+
+		checkbox.checkboxradio( "destroy" );
+		checkbox.checkboxradio({
+			disabled: null
+		});
+
+		strictEqual( widget.hasClass( "ui-state-disabled" ), true,
+			"passing null to disabled on startup checks the dom" );
+		strictEqual( checkbox.is( ":disabled" ), true,
+			"passing null to disabled on startup checks the dom" );
+	});
+	test( "options: icon", function() {
+		var checkbox = $( "#checkbox-option-icon" ),
+			widget;
+
+		expect( 9 );
+
+		checkbox.checkboxradio({ icon: false });
+
+		widget = checkbox.checkboxradio( "widget" );
+
+		strictEqual( widget.find( "span" ).length, 0,
+			"Label does not contain a span when created with icon:false" );
+
+		checkbox.checkboxradio( "destroy" );
+
+		checkbox.checkboxradio({
+			icon: true
+		});
+
+		strictEqual( widget.find( "span" ).length, 1,
+			"Label contains a span when created with icon:true" );
+		strictEqual( widget.find( "span" ).is( ".ui-checkboxradio-icon.ui-corner-all.ui-icon." +
+			"ui-icon-background.ui-icon-blank" ), true,
+			"Icon span has proper classes when created not checked" );
+
+		checkbox.checkboxradio( "destroy" ).prop( "checked", true );
+
+		checkbox.checkboxradio({
+			icon: true
+		});
+
+		strictEqual( widget.find( "span" ).is( ".ui-checkboxradio-icon.ui-corner-all.ui-icon." +
+			"ui-icon-background.ui-icon-check" ), true,
+			"Icon span has proper classes when created checked" );
+
+		checkbox.checkboxradio( "option", "icon", false );
+
+		strictEqual( widget.find( "span" ).length, 0,
+			"Label does not contain a span when option set to icon:false and checked" );
+
+		checkbox.checkboxradio( "option", "icon", true );
+
+		strictEqual( widget.find( "span" ).is( ".ui-checkboxradio-icon.ui-corner-all.ui-icon." +
+			"ui-icon-background.ui-icon-check" ), true,
+			"Icon span has proper classes when option set to true and :is( checked )" );
+
+		checkbox.prop( "checked", false ).checkboxradio( "refresh" );
+		checkbox.checkboxradio( "option", "icon", false );
+
+		strictEqual( widget.find( "span" ).length, 0,
+			"Label does not contain a span when option set to icon:false and not checked" );
+
+		checkbox.checkboxradio( "option", "icon", true );
+
+		strictEqual( widget.find( "span" ).is( ".ui-checkboxradio-icon.ui-corner-all.ui-icon." +
+			"ui-icon-background.ui-icon-blank" ), true,
+			"Icon span has proper classes when option set to true and not checked" );
+
+		checkbox.checkboxradio( "destroy" );
+
+		strictEqual( widget.find( "span" ).length, 0,
```

Comment:
Let's drop this, the destroy method test already covers it. Or if it doesn't extend it.